tools/codegen/dest/lazy_ir.py

```diff
@@ -58,6 +73,37 @@ def aten_symbol(schema: LazyIrSchema) -> str:
         return f'c10::Symbol::fromQualString("aten::{schema.aten_name}")'
     return f'at::aten::{schema.aten_name}'
 
+def cached_shape_inference(func: NativeFunction, schema: LazyIrSchema,
```

Comment:
~~This code guarded by `shape_cache->Get(dag_hash)` somehow? I.e. we wouldn't call into meta kernels if we already have the shapes for a given `dag_hash`?~~

found it! 

```
        auto cached_shapes = shape_cache->Get(dag_hash);
        if (cached_shapes == nullptr) {{{shapes_body}
```